Formula/r/recc.rb

```diff
@@ -0,0 +1,231 @@
+      # 4. ~/.recc/recc.conf
+      # 5. <%= "#{etc}/recc/recc.conf" %>
+      # 6. <%= "#{prefix}/etc/recc/recc.conf" %>
+      ##
+
+      # Protocol version used for communicating with the REAPI server
+      reapi_version=2.2
+
+      # The URI of the server
+      server=<%= "unix://#{var}/recc/casd/casd.sock" %>
+      # Instance name provided by the server
+      instance=recc-server
+
+      # Identifier to correlate all recc invocations#{" "}
+      correlated_invocations_id=<%= #{`hostname`.strip} %>
+
+      # Use cache-only mode, build anything not available in the cache locally
+      cache_only=1
+      # Upload results from the local builds to cache
+      cache_upload_local_build=1
+      # Upload failed results from the local builds to cache
+      cache_upload_failed_build=1
+
+      # Enable for link commands#{" "}
+      link=1
+      # Use cache-only mode for link commands#{" "}
+      link_cache_only=1
+
+      # Report all entries returned by the dependency command, even if they are absolute paths
+      deps_global_paths=1
+      # Do not rewrite absolute paths to be relative.
+      no_path_rewrite=1
+
+      # Do not retry#{" "}
+      retry_limit=0
+
+      # Platform properties
+      REMOTE_PLATFORM_OS=<%= platform_os %>
+      REMOTE_PLATFORM_ISA=<%= Hardware::CPU.arch.to_s %>
+    EOS
+
+    config_directory.mkpath
+    File.write(config_directory / "recc.conf", ERB.new(recc_conf_tmpl).result(binding))
+  end
+
+  def generate_recc_server(bin_directory)
+    (bin_directory / "recc-server").write <<~EOS
+      #!/usr/bin/env sh
+      #{bin}/buildbox-casd \
+        --local-server-instance=recc-server \
+        --protect-session-blobs \
+        #{var}/recc/casd
+    EOS
+  end
+
+  def generate_recc_wrapper(bin_directory, compiler)
+    (bin_directory / "recc-#{compiler}").write <<~EOS
+      #!/usr/bin/env sh
+      #{bin}/recc $(command -v #{compiler}) $@
+    EOS
+  end
+
+  def buildbox_cmake_args
+    %W[
+      -DCASD=ON
+      -DCASD_BUILD_BENCHMARK=OFF
+      -DCASDOWNLOAD=OFF
+      -DCASUPLOAD=OFF
+      -DFUSE=OFF
+      -DLOGSTREAMRECEIVER=OFF
+      -DLOGSTREAMTAIL=OFF
+      -DOUTPUTSTREAMER=OFF
+      -DRECC=ON
+      -DREXPLORER=OFF
+      -DRUMBA=OFF
+      -DRUN_BUBBLEWRAP=OFF
+      -DRUN_HOSTTOOLS=ON
+      -DRUN_OCI=OFF
+      -DRUN_USERCHROOT=OFF
+      -DTREXE=OFF
+      -DWORKER=OFF
+      -DRECC_CONFIG_PREFIX_DIR=#{etc}/recc
+    ]
+  end
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *buildbox_cmake_args, *std_cmake_args
+    system "cmake", "--build", "build", "-j", ENV.make_jobs.to_s
+    system "cmake", "--install", "build"
+
+    generate_recc_wrapper bin, "cc"
+    generate_recc_wrapper bin, "c++"
+    generate_recc_wrapper bin, "gcc"
+    generate_recc_wrapper bin, "g++"
+    generate_recc_wrapper bin, "clang"
+    generate_recc_wrapper bin, "clang++"
+
+    generate_recc_server bin
+
+    generate_recc_conf prefix / "etc/recc/"
```

Comment:
```suggestion
    generate_recc_conf prefix/"etc/recc/"
```
Note that writing this to `prefix/"etc"` means that users will not be able to write their own config files to `etc/"recc"`.